File: src/main/java/com/angular/service/PostsService.java
package com.angular.service;

import java.util.List;
import java.util.Map;

import com.alibaba.fastjson.JSONObject;
import com.angular.entity.Posts;
import com.angular.entity.User;

public interface PostsService {
    public Posts findById(int id);
    public List<Posts>  lists();
    public void edit(Posts posts);
    public void delete(int id);
}

File: angular/src/app/service/posts.service.ts
import { Injectable } from '@angular/core';
import {DataService} from './data.service';
import {Observable} from 'rxjs';
import { Configuration } from '../shared/configuration';

@Injectable({
  providedIn: 'root'
})
export class PostsService {
    apiUrl: string;

    constructor(private config: Configuration,
        private data: DataService) {
        this.apiUrl = config.API_URL;
    }

    addPosts(data: any): Observable<any> {
        return this.data.post<any>(this.apiUrl + 'posts/add', data);
    }

    editPosts(data: any): Observable<any> {
        return this.data.post<any>(this.apiUrl + 'posts/edit', data);
    }

    deletePosts(id: number): Observable<any> {
        let data = {
            id : id,
        };

        return this.data.post<any>(this.apiUrl + 'posts/delete', data);
    }
}
